python/ray/serve/tests/test_router.py

```diff
@@ -186,6 +188,12 @@ async def test_shard_key(serve_instance, task_runner_mock_actor):
 
 
 async def test_router_use_max_concurrency(serve_instance):
+    # The VisibleRouter::get_queues method needs to pickle queries
```

Comment:
Like I mentioned in the other PR, can we just remove the async_future member so we don't need these custom serializers?